src/conversion.jl

```diff
@@ -40,33 +41,65 @@ GI.getfeature(::Any, d::Data, i) = d[i,:]
 # Convert geometries to Meshes.jl types
 # --------------------------------------
 
-coords2point(coords) = Point(coords)
+function topoints(geom, is3d::Bool)
+  if is3d
+    [Point(GI.x(p), GI.y(p), GI.z(p)) for p in GI.getpoint(geom)]
+  else
+    [Point(GI.x(p), GI.y(p)) for p in GI.getpoint(geom)]
+  end
+end
+
+tochain(geom, is3d::Bool) = Chain(topoints(geom, is3d))
+
+function topolygon(geom, is3d::Bool)
+  exterior = topoints(GI.getexterior(geom), is3d)
+  if GI.nhole(geom) == 0
+    PolyArea(exterior)
+  else
+    holes = map(g -> topoints(g, is3d), GI.gethole(geom))
+    PolyArea(exterior, holes)
+  end
+end
+
+function GI.convert(::Type{Point}, ::GI.PointTrait, geom)
+  if GI.is3d(geom)
+    Point(GI.x(geom), GI.y(geom), GI.z(geom))
+  else
+    Point(GI.x(geom), GI.y(geom))
+  end
+end
+
+GI.convert(::Type{Segment}, ::GI.LineTrait, geom) = Segment(topoints(geom, GI.is3d(geom)))
 
-coords2multipoint(coords) = Multi(coords2point.(coords))
+GI.convert(::Type{Chain}, ::GI.LineStringTrait, geom) = tochain(geom, GI.is3d(geom))
 
-coords2chain(coords) = Chain(Point.(coords))
+GI.convert(::Type{Polygon}, ::GI.PolygonTrait, geom) = topolygon(geom, GI.is3d(geom))
 
-coords2multichain(coords) = Multi(coords2chain.(coords))
+function GI.convert(::Type{Multi}, ::GI.MultiPointTrait, geom)
+  Multi(topoints(geom, GI.is3d(geom)))
```

Comment:
Should we follow the pattern in the other Multi methods and call is3d only once here as well?